main.go

```diff
@@ -34,32 +35,36 @@ func main() {
 
 func run(ctx context.Context) error {
 	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, os.Kill)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 	svcErrors := make(chan error, 1)
 
 	// Read config
 	cfg, err := config.Get()
 	if err != nil {
 		return fmt.Errorf("unable to retrieve service configuration: %w", err)
 	}
+	log.Info(ctx, "config on startup", log.Data{"config": cfg, "build_time": BuildTime, "git-commit": GitCommit})
+
+	serviceCtx, cancel := context.WithCancel(ctx)
```

Comment:
I wonder if it's possible to reuse the ctx variable like `ctx, cancel = context.Cancel(ctx)` here, the way you might do `i = i +1`?

Reply:
The intention was to only use the `serviceCtx` for the service. But `ctx` is only used for logging, so we could as well rename it as you mention, and use it for everything